File: udgaard/src/main/java/com/skrymer/udgaard/repository/MarketBreadthRepository.java
package com.skrymer.udgaard.repository;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;

import com.skrymer.udgaard.model.MarketBreadth;
import com.skrymer.udgaard.model.MarketSymbol;
import com.skrymer.udgaard.model.Stock;

public interface MarketBreadthRepository extends MongoRepository<MarketBreadth, String> {
    Optional<Stock> findBySymbol(MarketSymbol symbol);
}

File: udgaard/src/main/kotlin/com/skrymer/udgaard/repository/MarketBreadthRepository.kt
package com.skrymer.udgaard.repository

import com.skrymer.udgaard.model.MarketBreadth
import com.skrymer.udgaard.model.MarketSymbol
import org.springframework.data.mongodb.repository.MongoRepository

interface MarketBreadthRepository : MongoRepository<MarketBreadth, MarketSymbol>
